app/routes/users/components/UserSettingsOAuth2Form.tsx

```diff
@@ -32,9 +32,9 @@ const UserSettingsOAuth2Form = () => {
   const { applicationId } = useParams<{ applicationId?: string }>();
   const isNew = applicationId === undefined;
   const application = useAppSelector((state) =>
-    selectOAuth2ApplicationById(state, {
-      applicationId,
-    }),
+    applicationId
+      ? selectOAuth2ApplicationById(state, applicationId)
+      : undefined,
```

Comment:
Wouldn't it be "safer" and cleaner if this was just handled by `selectOAuth2ApplicationById`?

Reply:
yea, I suppose. But it's kind of pain to do that. The "ById"-selectors comes directly from redux-toolkits `entityAdapter`, and don't support selecting by undefined id.

Reply:
It wasn't that bad, but it required some typecasting though. I added this and some more typing features in this PR: https://github.com/webkom/lego-webapp/pull/4604